### DISCUSSION TITLE
Identyfikator paczki niedostępny w API

### DISCUSSION THREAD

**Discussion by hahyes** [allegro/allegro-api#5436] [Suggestions - rejected]
Witam,

natknąłem się na taką sytuację, gdzie w wyniku błędu nie zapisały się identyfikatory niektórych przesyłek nadanych przez API Wysyłam z Allegro. Problem polega na tym, że te ID są potrzebne do pobrania szczegółów przesyłki czy pobrania etykiety, a API tego ID nie udostępnia w żaden sposób. Chyba, że gdzieś nie szukałem ale przegrzebałem dokumentację API i nie ma zasobu posiadającego to ID, abym mógł podejrzeć je i ewentualnie zaktualizować dane.

Sugerowane rozwiązania:
- `parcelId` w danych zamówienia - moim zdaniem najprostszy sposób
- możliwość drukowania etykiet za pomocą numeru przesyłki firmy kurierskiej - podejrzewam, że byłby dużo bardziej wymagający w implementacji, dodanie możliwości podania `waybill` do wyszukania odpowiedniej etykiety

Jestem zdziwiony, że tego ID nie ma w żaden sposób w API. Utrudnia to pobranie danych o stworzonej przesyłce. Gdybym na przykład chciał też synchronizować dane w bazie z paczkami tworzonymi w panelu Wysyłam z Allegro to bym prawdopodobnie nie był w stanie tego zrobić, ponieważ nie mógłbym pobrać szczegółów i etykiety, gdyby zaszła taka potrzeba.

Liczę na odpowiedź i pochylenie się nad problemem.

**Comment by PrzemyslawLukanowski** [comment]
Jeśli zapisujesz po swojej strony numery UUID za pomocą, których tworzysz przesyłki, to `parcelId `otrzymasz przy odpytaniu o status utworzenia paczki - [GET /parcel-management/parcel-create-commands/{commandId}](https://developer.allegro.pl/wza/#jak-sprawdzi%c4%87-status-utworzenia-paczki). Czy takie rozwiązanie jest dla Ciebie wystarczające?

**Reply by hahyes** [reply]
@PrzemyslawLukanowski Jak mówiłem, w wyniku błędu nie zapisały się te informacje, więc to nie zadziała. Nie mając UUID, numer `parcelId` przepada bezpowrotnie. Ponadto przykład z synchronizacją ręcznie tworzonych w Wysyłam z Allegro przesyłek dalej obowiązuje - nie mogę pobrać szczegółów czy etykiety takiej paczki, bo API nie udostępnia nigdzie danych na to pozwalających. Nie ma też żadnej listy utworzonych przesyłek, z której mógłbym zaktualizować dane. Jakby nie patrzeć, jestem w kropce. 

Chyba, że istnieje jakieś rozwiązanie, o którym jeszcze nie wiem.

**Reply by PrzemyslawLukanowski** [reply]
Niestety, ale inny sposób nie istnieje. Pozostawiam wątek w Discussions, tak aby rownież inni użytkownicy mogli zapoznać się z tą potrzebą.

**Reply by Jur3k** [reply]
zakładam że Allegro zapisuje te UUID, bo w jakiś sposób udostępnia pod https://allegro.pl/moje-allegro/sprzedaz/zamowienia/{id zamowienia}?sellerId={id sprzedawcy} dane o przesyłce
szkoda że metoda https://api.allegro.pl/order/checkout-forms/{id zamowienia}/shipments nie zwraca tego UUID albo parcelId

**Comment by robsonek** [comment]
Widzę że temat od dawna stoi w miejscu. Zgłaszam także zapotrzebowanie na taką metodę.

**Comment by arekpietrzyk** [comment]
Widzę, że to rzadka potrzeba, ale trafiłem tu właśnie w celu zgłoszenia tego samego tematu.

W momencie kiedy przesyłek nie tworzymy przez API, a jedynie przez Panel Allegro, niemożliwym jest post-processing danych przez API. Nie ma skąd pobrać shipmentId.

Proponuję, aby to ID znalazło się w odpowiedzi na request do `GET /order/checkout-forms/{uuid}/shipments`

**Reply by arekpietrzyk** [reply]
Pozwolę sobie wywołać @PrzemyslawLukanowski, aby zwrócić uwagę na dość archiwalny wątek :-)

**Reply by MaciejFrackowiak** [reply]
@arekpietrzyk Na chwilę obecną nie planujemy wprowadzać zmiany w tym zakresie.

**Reply by arekpietrzyk** [reply]
@MaciejFrackowiak a chociaż jakiś endpoint aby poprać listę przesyłek zleconych w ramach WzA ?

**Reply by MaciejFrackowiak** [reply]
Nie planujemy udostępniać dodatkowego endpointu do zwracania grupowej listy wygenrowanych przesyłek.

**Reply by kolszowskiEUP** [reply]
@MaciejFrackowiak U mnie też zaistniała sytuacja, w której potrzebuję pobrać etykietę przewozową stworzoną przez panel Allegro, ale nie jest to możliwe przez brak możliwości wyciągnięcia shipmentId. Podpinam się pod prośbę o dodanie tego do API. Tak jak wspomniano wcześniej, dodatkowe pole w `GET /order/checkout-forms/{uuid}/shipments` byłby do tego raczej wygodnym rozwiązaniem.
